samples/Microsoft.Bot.Samples.Ai.QnA.Translator/Startup.cs

```diff
@@ -0,0 +1,178 @@
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+            Configuration = builder.Build();
+        }
+
+        public IConfigurationRoot Configuration { get; }
+        private void SetLanguage(ITurnContext context, string language) => context.GetConversationState<CurrentUserState>().Language = language;
+        private void SetLocale(ITurnContext context, string locale) => context.GetConversationState<CurrentUserState>().Locale = locale;
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddBot<QnAMakerBot>(options =>
+            {
+                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+
+                var qnaOptions = new QnAMakerMiddlewareOptions
+                {
+                    // add subscription key and knowledge base id
+                    SubscriptionKey = "xxxxxx",
+                    KnowledgeBaseId = "xxxxxx"
+                };
+
+                var middleware = options.Middleware;
+                Dictionary<string, List<string>> patterns = new Dictionary<string, List<string>>();
+                patterns.Add("fr", new List<string> { "mon nom est (.+)" });//single pattern for fr language
+                middleware.Add(new ConversationState<CurrentUserState>(new MemoryStorage()));
+                middleware.Add(new TranslationMiddleware(new string[] { "en" }, "<your translator key here>", patterns, GetActiveLanguage, SetActiveLanguage));
+                middleware.Add(new LocaleConverterMiddleware(GetActiveLocale, SetActiveLocale, "en-us", new LocaleConverter()));
+                middleware.Add(new QnAMakerMiddleware(qnaOptions));
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseDefaultFiles()
+                .UseStaticFiles()
+                .UseBotFramework();
+        }
+        protected bool IsSupportedLanguage(string language) => _supportedLanguages.Contains(language);
+        protected async Task<bool> SetActiveLanguage(ITurnContext context)
+        {
+            bool changeLang = false;//logic implemented by developper to make a signal for language changing 
+            //use a specific message from user to change language
+            var messageActivity = context.Activity.AsMessageActivity();
+            if (messageActivity.Text.ToLower().StartsWith("set my language to"))
```

Comment:
This will often explode, for instance if the emulator is connecting. 

AsMessageActivity() will return NULL on any Activity other than message, such as a ConversationJoin Activity. This code will then throw an exception.

Reply:
Fixed.